### PR TITLE
refactor: Introduce administrative permission service

### PR DESCRIPTION
**Pull request by seakayone**
- **refactor: Rename PermissionResponder and add AdministrativePermissionService**
- **Integrate AdministrativePermissionService into PermissionsResponder and PermissionsRestService**
- **disable login for builtIn users**
- **fmt**
- **refactor: make org.knora.webapi.responders.admin.PermissionsResponder#permissionsDataGetADM accept only a KnoraUser**
- **simplify**
- **rename getPermissionData**
- **remove PermissionDataGetADM**
- **move code which calculates User permissions from PermissionsResponder to KnoraUserToUserConverter**

### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)
- [ ] deprecated: Deprecation warning (ideally referencing a migration guide)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/slice/admin/api/AdminApiModule.scala [hunk 1]
```diff
@@ -38,10 +39,10 @@ import org.knora.webapi.store.triplestore.api.TriplestoreService
 object AdminApiModule {
 
   type Dependencies =
-    AppConfig & AssetPermissionsResponder & AuthorizationRestService & BaseEndpoints & CacheService &
+    AppConfig & AdministrativePermissionService & AssetPermissionsResponder & AuthorizationRestService & BaseEndpoints & CacheService &
       GroupsResponderADM & GroupService & HandlerMapper & KnoraProjectService & KnoraResponseRenderer &
       KnoraUserService & KnoraUserToUserConverter & ListsResponder & MaintenanceService & OntologyCache &
-      PasswordService & PermissionsResponderADM & ProjectExportService & ProjectImportService & ProjectService &
+      PasswordService & PermissionsResponder & ProjectExportService & ProjectImportService & ProjectService &
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/api/AdminApiModule.scala:45]: as above

**Review by BalduinLandolt — APPROVED** [review]
LGTM

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/GroupService.scala [hunk 2]
```diff
@@ -39,16 +36,6 @@ final case class GroupService(
...
-
-  def toKnoraGroup(group: Group): KnoraGroup =
-    KnoraGroup(
-      id = GroupIri.unsafeFrom(group.id),
-      groupName = GroupName.unsafeFrom(group.name),
-      groupDescriptions = GroupDescriptions.unsafeFrom(group.descriptions),
-      status = GroupStatus.from(group.status),
-      belongsToProject = group.project.map(it => ProjectIri.unsafeFrom(it.id)),
-      hasSelfJoinEnabled = GroupSelfJoin.from(group.selfjoin),
-    )
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/GroupService.scala:51]: Could you bring it back please? It is unused not, but will be soon - group slice WIP.

**Review by mpro7 — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/GroupService.scala (hunk 2 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/GroupService.scala:51]: Please readd it yourself when in use and ideally tested.

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserToUserConverter.scala [hunk 3]
```diff
@@ -42,7 +43,94 @@ final case class KnoraUserToUserConverter(
...
+
+    /* Get all permissions per project, applying permission precedence rule */
+    def calculatePermission(
+      projectIri: ProjectIri,
+      extendedUserGroups: Seq[GroupIri],
+    ): Task[(ProjectIri, Set[PermissionADM])] = {
+      /* Follow the precedence rule:
+         1. ProjectAdmin > 2. CustomGroups > 3. ProjectMember > 4. KnownUser
+         Permissions are added following the precedence level from the highest to the lowest. As soon as one set
+         of permissions is written into the buffer, any additionally permissions do not need to be added. */
```
[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserToUserConverter.scala:92]: TBH, I was even considering to remove it. However it explains why the implementation below is how it is and I think that is the right place.

**Pull request merged**